### PR TITLE
add deserialize_file()

### PR DESCRIPTION
**Pull request by NeroBlackstone**
Related issue #46.

I did not write test code, since I don't know how to build a download server #50 

But codes should work, I test it on another project.

image

### CODE REVIEW

File: src/client.jl [hunk 1]
```diff
@@ -599,4 +600,49 @@ is_request_interrupted(ex::TaskFailedException) = is_request_interrupted(ex.task
...
+    rename_file::String="",
+    overwrite::Bool=true
+    )::Tuple{Any,ApiResponse,String}
+    
+    result, http_response = api_call()
+
+    content_disposition_str = OpenAPI.Clients.header(http_response.raw,"content-disposition","")
+    content_type_str = OpenAPI.Clients.header(http_response.raw,"content-type","")
+    
+    file_name = if length(rename_file) > 0
```
Comment (tanmaykm) [inline src/client.jl:630]:
```suggestion
    file_name = if !isempty(rename_file)
```

File: src/client.jl [hunk 2]
```diff
@@ -599,4 +600,49 @@ is_request_interrupted(ex::TaskFailedException) = is_request_interrupted(ex.task
...
+        rename_file
+    elseif length(content_disposition_str) > 0
```
Comment (tanmaykm) [inline src/client.jl:632]:
```suggestion
    elseif !isempty(content_disposition_str)
```

File: src/client.jl [hunk 3]
```diff
@@ -599,4 +600,49 @@ is_request_interrupted(ex::TaskFailedException) = is_request_interrupted(ex.task
...
+        content_disposition_str
+    else 
+        "response"*extension_from_mime(MIME(content_type_str))
+    end
+
+    file_path = joinpath(mkpath(folder_path),file_name)
+
+    if !(isfile(file_path)&&!overwrite)
```
Comment (tanmaykm) [inline src/client.jl:640]: Not sure if we should accept an option to not `overwrite`. There may be more than one way the method may be expected to behave when file already exists and overwrite is false. Same goes for the folder name.

I feel it is best left for the caller to pass the exact folder and file names, and let the method always replace the file.

**Review by tanmaykm — CHANGES_REQUESTED** [review]

**Comment by NeroBlackstone** [comment]
I think I have finished what you suggested.
btw, Shall we add a `Configuration` struct to save server/client configuration like [here](https://github.com/Kaggle/kaggle-api/blob/75dfa27d3c2886ea46e4d0c6bfcc0cc322af554b/kaggle/configuration.py#L56)?

And `deserialize_file()` uses the default download location from this object.
So we can change all default download locations in one place, instead manually passing argument again and again when using `deserialize_file()`.

**Review by tanmaykm — APPROVED** [review]

**Comment by tanmaykm** [comment]
I think the configuration struct is something specific that a client application needs to maintain, it should not be part of the openapi package.

Merging this now, and I'll attempt writing some test cases for it subsequently.

**Comment by NeroBlackstone** [comment]
Thanks, after testing these codes and adding docs, please make a new release. 
I'll close #46 if everything is okay.

**Pull request merged**